hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/DroolsRulesEngineImpl.java

```diff
@@ -119,7 +119,7 @@ public void addData(Data data) {
     }
 
     @Override
-    public void addData(Collection<Data> data) {
+    public void addData(TreeSet<Data> data) {
```

Comment:
Minor.
Same here.
I don't have any preference if we maintain Collection<Data> or the TreeSet<Data> in the signature, but as it is `@Override` it should be changed in the interface also.